kubernetes-client-openapi/src/main/java/io/micronaut/kubernetes/client/openapi/credential/ExecCommandCredentialLoader.java

```diff
@@ -16,30 +16,34 @@
 package io.micronaut.kubernetes.client.openapi.credential;
 
 import io.micronaut.context.annotation.BootstrapContextCompatible;
+import io.micronaut.core.annotation.Internal;
 import io.micronaut.json.JsonMapper;
-import io.micronaut.kubernetes.client.openapi.config.KubernetesClientConfiguration;
+import io.micronaut.kubernetes.client.openapi.config.KubeConfig;
+import io.micronaut.kubernetes.client.openapi.config.KubeConfigLoader;
 import io.micronaut.kubernetes.client.openapi.config.model.ExecConfig;
 import io.micronaut.kubernetes.client.openapi.config.model.ExecEnvVar;
-import io.micronaut.kubernetes.client.openapi.config.KubeConfig;
 import io.micronaut.kubernetes.client.openapi.credential.model.ExecCredential;
 import jakarta.inject.Singleton;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.io.File;
 import java.io.InputStream;
+import java.nio.file.Path;
 import java.time.ZoneId;
 import java.time.ZonedDateTime;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
 
 /**
  * The credential loader which uses the exec command from the kube config to get credentials.
  */
 @Singleton
 @BootstrapContextCompatible
-class ExecCommandCredentialLoader implements KubernetesCredentialLoader {
+@Internal
+public class ExecCommandCredentialLoader implements KubernetesCredentialLoader {
```

Comment:
is this public for a reason? If so document why

Reply:
If it is not public it can't be replaced with custom implementation
```
@Replaces(ExecCommandCredentialLoader.class)
```